### ISSUE TITLE
Determine attachment content_type from filename extension

### ISSUE THREAD

**Issue by mkoeppe** [sagemath/trac-to-github#104]
Currently we use `application/octet-stream` for everything, but the importer complains:
```
now="2023-01-10T14:38:37Z" guid=cae4a192-8fed-11ed-84b0-230110063154 command=import fn="GitHub::Migrator::AttachmentImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=github/migrator/attachment_importer/user_asset_extended_content_types model_id=1 field=name validation_error="has a file extension that does not match the content type"
```

**Comment by mkoeppe** [comment]
using https://docs.python.org/3/library/mimetypes.html

**Comment by mkoeppe** [comment]
Same problem with `repository_file`
```
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=2 field=name validation_error="has a file extension that does not match the content type"
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=3 field=name validation_error="has a file extension that does not match the content type"
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=4 field=content_type validation_error="is not included in the list"
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=4 field=name validation_error="has a file extension that does not match the content type"
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=5 field=content_type validation_error="is not included in the list"
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=5 field=name validation_error="has a file extension that does not match the content type"
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=6 field=content_type validation_error="is not included in the list"
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=6 field=name validation_error="has a file extension that does not match the content type"
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=7 field=content_type validation_error="is not included in the list"
now="2023-01-10T21:25:11Z" guid=cae4a192-8fed-11ed-84b0-230110131425 command=import fn="GitHub::Migrator::RepositoryFileImporter#import_model" class="ActiveRecord::RecordInvalid" model_name=repository_file model_id=7 field=name validation_error="has a file extension that does not match the content type"
```

**Comment by mkoeppe** [comment]
Fixed by 14edd3e
